src/H5FDros3.h

```diff
@@ -84,26 +135,94 @@ typedef struct H5FD_ros3_fapl_t {
     char    secret_key[H5FD_ROS3_MAX_SECRET_KEY_LEN + 1];
 } H5FD_ros3_fapl_t;
 
+/**
+ * \struct H5FD_ros3_fapl_ext_t
+ * \brief Extended configuration structure for holding the configuration data
+ *        to the #H5FD_ROS3 driver via a File Access Property List.
+ *
+ * \details H5FD_ros_fapl_ext_t is a public structure that is used to pass
+ *          configuration data to the #H5FD_ROS3 driver via a File Access
+ *          Property List.
+ *
+ * \var H5FD_ros3_fapl_t H5FD_ros3_fapl_ext_t::fa
+ *      Configuration structure for H5Pset_fapl_ros3() / H5Pget_fapl_ros3().
+ *
+ * \var char H5FD_ros3_fapl_ext_t::token[H5FD_ROS3_MAX_SECRET_TOK_LEN + 1]
+ *      A string which specifies the session/security token.
+ *
+ */
+typedef struct H5FD_ros3_fapl_ext_t {
+    H5FD_ros3_fapl_t fa;
+    char             token[H5FD_ROS3_MAX_SECRET_TOK_LEN + 1];
+} H5FD_ros3_fapl_ext_t;
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * \brief Internal routine to initialize #H5FD_ROS3 driver. Not meant to be
+ *        called directly by an HDF5 application.
+ */
 H5_DLL hid_t H5FD_ros3_init(void);
 
 /**
  * \ingroup FAPL
  *
- * \todo Add missing documentation
+ * \brief Queries a File Access Property List for #H5FD_ROS3 file driver properties.
+ *
+ * \fapl_id
+ * \param[out] fa_out Pointer to #H5FD_ROS3 driver configuration structure.
+ * \returns \herr_t
  */
 H5_DLL herr_t H5Pget_fapl_ros3(hid_t fapl_id, H5FD_ros3_fapl_t *fa_out);
 
 /**
  * \ingroup FAPL
  *
- * \todo Add missing documentation
+ * \brief Modifies the specified File Access Property List to use the #H5FD_ROS3 driver.
+ *
+ * \fapl_id
+ * \param[in] fa Pointer to #H5FD_ROS3 driver configuration structure.
+ * \returns \herr_t
  */
 H5_DLL herr_t H5Pset_fapl_ros3(hid_t fapl_id, const H5FD_ros3_fapl_t *fa);
 
+/**
+ * \ingroup FAPL
+ *
+ * \brief Queries a File Access Property List for #H5FD_ROS3 file driver session/security
+ *        token.
+ *
+ * \fapl_id
+ * \param[in] size Size of the provided char array for storing the session/security token.
+ * \param[out] token Session/security token.
+ * \returns \herr_t
+ *
+ * \since 1.14.2
+ */
+H5_DLL herr_t H5Pget_fapl_ros3_token(hid_t fapl_id, size_t size, char *token);
+
+/**
+ * \ingroup FAPL
+ *
+ * \brief Modifies the specified File Access Property List to use the #H5FD_ROS3 driver
+ *        by adding the specified session/security token.
+ *
+ * \fapl_id
+ * \param[in] token Session/security token.
+ * \returns \herr_t
+ *
+ * \details H5Pset_fapy_ros3_token() modifies the File Access Property List to use the
+ *          #H5FD_ROS3 driver by adding or updating the session/security token.
+ *
+ *          The session token is only needed when you want to access a S3 bucket
+ *          using temporary security credentials.
```

Comment:
For the brief and details sections here, I'd update to just mention that they set the session/security token, since `H5Pset_fapl_ros3` is the API that actually modifies the FAPL to use the ROS3 VFD.